### PR TITLE
Add sample data script and corresponding .rda files

### LINKED ISSUE #4: Set up and document example data objects based on `example-complex-forecast-hub`
These are the data objects described under "Example 2" in [this discussion comment](https://github.com/orgs/Infectious-Disease-Modeling-Hubs/discussions/6#discussioncomment-8698190).

It seems this package should have at least the following two data objects providing example data related to forecasts:

- `forecast_outputs`: a `model_output_tbl` with model forecasts
- `forecast_target_ts`: a data frame with target data in time series format

Possibly, it may be helpful to also include a data frame of observed target values, perhaps called `forecast_target_values`.  We have discussed that we might set up functionality for computing these target values rather than storing them as part of a hub -- but I imagine that it could be helpful to just have them pre-computed and available as a data object in the package.

There is a discussion about what data to include in discussion #1

### PR DESCRIPTION
**Pull request by bsweger**
Resolves #4. 

This changeset generates sample data files using data from the complex-example-forecast-hub. For this first pass, the samples are generated from local files (i.e., a cloned copy of complex-example-forcast-hub). 

Once complex-example-forecast-hub is fully onboarded to the cloud, we'll pull from there instead.

### CODE REVIEW

**Comment by elray1** [comment]
R CMD check is demanding that we document our data sets. I think we could pull from [here](https://github.com/Infectious-Disease-Modeling-Hubs/hubEnsembles/blob/main/R/data.R), with some light adaptations. Should I do that, @bsweger, or would you like to?

**Comment by bsweger** [comment]
> R CMD check is demanding that we document our data sets. I think we could pull from [here](https://github.com/Infectious-Disease-Modeling-Hubs/hubEnsembles/blob/main/R/data.R), with some light adaptations. Should I do that, @bsweger, or would you like to?

I can take a stab at it, but you will likely have some wording revisions. Does that information go at the top of `generate_example_forecast_data.R`?

**Comment by elray1** [comment]
It should go in a file called `data.R` in an `R` directory in this repository. Here's the description of how to document data sets in the R Packages book: https://r-pkgs.org/data.html#sec-documenting-data

**Review by elray1 — CHANGES_REQUESTED** [review]
Adding an actual/formal review to request 2 changes:

- [x] Add documentation for the data objects.  As (partially) discussed in some misc. comments, this will involve adding a `data.R` file that documents the data objects and also running `devtools::document()` to add the documentation to the R package.
- [ ] Since we did this, we've decided on two other minor updates to the target values data format.  It would be nice to get those updates in this PR as well.  Those changes would be made first in the example-complex-forecast-hub repository with the source data, see [this issue](https://github.com/Infectious-Disease-Modeling-Hubs/example-complex-forecast-hub/issues/25).

**Comment by bsweger** [comment]
> * umentation for the data objects.  As (partially) discussed in some misc. comments, this will involve adding a `data.R` file that documents the data objects and also running `devtools::document()` to add the documentation to the R package.
> 
> * [ ]  Since we did this, we've decided on two other minor updates to the target values data format.  It would be nice to get those updates in this PR as well.  Those changes would be made first in the example-complex-forecast-hub repository with the source data, see [this issue](https://github.com/Infectious-Disease-Modeling-Hubs/example-complex-forecast-hub/issues/25).

@elray1 just pushed a commit for the first note, and I'm expecting to do some updates on that documentation based on your feedback

For the second issue, in the spirit of smaller changes, more quickly, my vote is to get this PR merged and then tackle the issue you linked as a separate piece of work. 

It's okay to change our minds/evolve, but I don't like keeping PRs open for long periods of time when we do so. It likely doesn't matter for `hubExample`, but if this repo was under active development with many devs, lingering PRs increase the liklihood of merge conflicts and other annoyances.

File: R/data.R [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#'   \item{target_end_date}{the date of occurrence of the outcome of interest in yyyy-mm-dd format;
+#'          this can be calculated directly from the `reference_date` and `horizon`
+#'          as follows: `target_end_date = reference_date + 7*horizon`}
+#'   \item{target}{a unique identifier for the target}
+#'   \item{output_type}{the type of representation of the prediction}
+#'   \item{output_type_id}{more identifying information specific to the output type;
+#'        output_type_id is not relevant for every kind of output_type (for example,
+#'        hubs will not expect output_type_id values when the output_type is mean or median}
+#'   \item{value}{the model’s prediction}
+#'   \item{model_id}{the name of the model}
```
Comment (bsweger) [inline R/data.R:22]: @elray1 question about `model_id`: is that a column we'd expect to see in a hub's model output data? I thought we derived it from the filename.

Comment (elray1) [reply R/data.R:33]: It is true that when the data are sitting in a hub, the `model_id` is encoded in the file name.  But when we `collect` the data into a data frame in a working R (or in the future, python) session, the `model_id` is added into the data.  And the intent of this example object is to represent what a user might get after running `collect_hub()`.  (Maybe we should say that in this documentation.)

**Review by elray1 — APPROVED** [review]

**Comment by elray1** [comment]
I'm approving this PR, with these notes:

 - We may want to make an update related to [this thread](https://github.com/Infectious-Disease-Modeling-Hubs/hubExamples/pull/8#discussion_r1544667339)
 - R CMD check is failing due to an unrelated issue which i've filed in #9 
 - I filed separate issue #10 for item 2 in my initial review.

@bsweger I'll leave it to you to do something about the first point here or not and then merge this?

File: R/data.R (hunk 1 above)
Comment (bsweger) [reply R/data.R:35]: Ah, makes sense--thank you for that clarification. Just pushed a commit with that note.

**Pull request merged**